redcap/project.py

```diff
@@ -192,6 +192,46 @@ def export_fem(self, arms=None, format='json', df_kwargs=None):
 
             return self.read_csv(StringIO(response), **df_kwargs)
 
+    def export_field_names(self, field=None, format='json', df_kwargs=None):
+        """
+        Export the project's field names
+
+        Parameters
+        ----------
+        fields : str
+            Limit exported field name to this field (only single field supported).
+            When not provided, all fields returned.
+        format : (``'json'``), ``'csv'``, ``'xml'``, ``'df'``
+            Return the metadata in native objects, csv or xml.
+            ``'df'`` will return a ``pandas.DataFrame``.
+        df_kwargs : dict
+            Passed to ``pandas.read_csv`` to control construction of
+            returned DataFrame.
+            by default ``{'index_col': 'original_field_name'}``
+
+        Returns
+        -------
+        metadata : list, str, ``pandas.DataFrame``
+            metadata sttructure for the project.
+        """
```

Comment:
1. I think we should be a little more specific about _which_ field names are being exported -- for example you could get the checkbox _export_ field names with this method
2. Small spelling correction 🔤 
```suggestion
        """
        Export the project's export field names

        Parameters
        ----------
        fields : str
            Limit exported field name to this field (only single field supported).
            When not provided, all fields returned.
        format : (``'json'``), ``'csv'``, ``'xml'``, ``'df'``
            Return the metadata in native objects, csv or xml.
            ``'df'`` will return a ``pandas.DataFrame``.
        df_kwargs : dict
            Passed to ``pandas.read_csv`` to control construction of
            returned DataFrame.
            by default ``{'index_col': 'original_field_name'}``

        Returns
        -------
        metadata : list, str, ``pandas.DataFrame``
            metadata structure for the project.
        """
```

Reply:
Thanks for the response. This package has been super helpful as I work on this project. I've never used REDCap before, and this has saved a bunch of headaches.

My thoughts for just returning the whole list of dicts:

1) The two fields do differ for multiple response checkboxes. You'd get something like:

```
[{'original_field_name': 'checkbox',
  'choice_value': '1',
  'export_field_name': 'checkbox___1'},
 {'original_field_name': 'checkbox',
  'choice_value': '2',
  'export_field_name': 'checkbox___2'}]
```

You might want to export different versions for different things. It's simple enough to do a list comprehension to grab what you want after it returns, and exposing the raw record gives the most flexibility.

2) Per @sburns comments on my earlier PR, I got the impression that the design philosophy was to put most of the onus on doing the processing on the user, as opposed to making those decisions for them.

That said, maybe a compromise approach would be to add a keyword flag that allows you to specify what you get in return: perhaps 'original' (which returns the list of strings of original field names, or perhaps a list of tuples containing the field name and the choice value), 'export' (which returns the list of strings of export field names), or 'raw' (which just returns the list of dicts) with the default value being set to one of your choice?

Reply:
Oh, my apologies if I gave the impression that I was looking for you to refactor your code 🙇‍♂️ 

I think the code is solid 💪 and I was just suggesting minor updates to the documentation 📝 ✔️